### PR TITLE
meson: dependency on ml-api should be optional @open sesame 03/31 17:58

### PR DESCRIPTION
**Pull request by myungjoo**
Users should be able to build nntrainer in a system without
ML-API and nnstreamer by default.

ML-API and nnstreamer dependency should only be mandated
in related systems (a few embedded systems).

Let's keep it "auto" so that most external users can forget
about nnstreamer and ML-API.

Addresses #1853

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -202,22 +223,7 @@ if not iniparser_dep.found()
...
-    warning('capi-nnstreamer version is too old, we do not know if it works with older nnstreamer version')
-  endif
-endif
-
-ml_api_common_dep = dummy_dep
-
-if get_option('platform') != 'android'
-  ml_api_common_dep = dependency(get_option('capi-ml-common-actual'), required: true)
-else
+if get_option('platform') == 'android'
```
Comment (jijoongmoon) [inline meson.build:226]: like the below, if we will download ml-api-inference, then, we need to set -DML_API_COMMON=1 for android.

**Comment by jijoongmoon** [comment]
LGTM, after android ci build test.

**Comment by jijoongmoon** [comment]
I think there are CI server issues. I will fix it and let you know.

**Comment by myungjoo** [comment]
Android build issue resolved. PTAL: @jijoongmoon @songgot @jihochu @lhs8928

**Review by jijoongmoon — APPROVED** [review]
CI issue is fixed.

**Pull request merged**